OptimizelySDK/Odp/OdpManager.cs

```diff
@@ -0,0 +1,274 @@
+        /// <param name="segmentsToCheck">Audience segments to consider</param>
+        /// <returns>True if settings were update otherwise False</returns>
+        public bool UpdateSettings(string apiKey, string apiHost, List<string> segmentsToCheck)
+        {
+            var newConfig = new OdpConfig(apiKey, apiHost, segmentsToCheck);
+            if (_odpConfig.Equals(newConfig))
+            {
+                return false;
+            }
+
+            _odpConfig = newConfig;
+
+            EventManager.Flush();
+            EventManager.UpdateSettings(_odpConfig);
+
+            SegmentManager.ResetCache();
+            SegmentManager.UpdateSettings(_odpConfig);
+
+            return true;
+        }
+
+        /// <summary>
+        /// Attempts to fetch and return a list of a user's qualified segments.
+        /// </summary>
+        /// <param name="userId">FS User ID</param>
+        /// <param name="options">Options used during segment cache handling</param>
+        /// <returns>Qualified segments for the user from the cache or the ODP server</returns>
+        public List<string> FetchQualifiedSegments(string userId, List<OdpSegmentOption> options)
+        {
+            if (!_enabled || SegmentManager == null || !_odpConfig.IsReady())
+            {
+                _logger.Log(LogLevel.ERROR, Constants.ODP_NOT_ENABLED_MESSAGE);
+                return null;
+            }
+
+            return SegmentManager.FetchQualifiedSegments(userId, options);
+        }
+
+        /// <summary>
+        /// Send identification event to ODP for a given full-stack User ID
+        /// </summary>
+        /// <param name="userId">User ID to send</param>
+        public void IdentifyUser(string userId)
+        {
+            if (!_enabled || EventManager == null || !_odpConfig.IsReady())
+            {
+                _logger.Log(LogLevel.DEBUG, "ODP identify event not dispatched (ODP disabled).");
+                return;
+            }
+
+            if (!EventManager.IsStarted)
+            {
+                _logger.Log(LogLevel.DEBUG,
+                    "ODP identify event not dispatched (ODP not integrated).");
+                return;
+            }
+
+            EventManager.IdentifyUser(userId);
+        }
+
+        /// <summary>
+        /// Add event to queue for sending to ODP
+        /// </summary>
+        /// <param name="type">Type of event (typically `fullstack` from server-side SDK events)</param>
+        /// <param name="action">Subcategory of the event type</param>
+        /// <param name="identifiers">Key-value map of user identifiers</param>
+        /// <param name="data">Event data in a key-value pair format</param>
+        public void SendEvent(string type, string action, Dictionary<string, string> identifiers,
+            Dictionary<string, object> data
+        )
+        {
+            if (!_enabled || EventManager == null || !_odpConfig.IsReady())
```

Comment:
This validations should be in a single method and return true or false. please refactor it.

Reply:
My copy-paste finger started hurting, and I thought of that too.  

L89, would have me injecting the Manager that should be checked for `null` so I just left it.

Let me play with some options....